engine/src/image.cpp

```diff
@@ -165,6 +165,8 @@ MCImage::MCImage(const MCImage &iref) : MCControl(iref)
 	m_locked_image = nil;
 	m_locked_bitmap = nil;
 	m_needs = nil;
+    
+    filename = MCValueRetain(kMCEmptyString);
```

Comment:
Just move filename = MCValueRetain(iref.filename) to here instead. The issue is that setbitmap() MCValueAssigns to filename, but it isn't valid until after it is called at present.